eva/parser/parser_visitor/_table_sources.py

```diff
@@ -33,6 +34,20 @@ def visitTableSources(self, ctx: evaql_parser.TableSourcesContext):
         for table_sources_index in range(table_sources_count):
             table = self.visit(ctx.tableSource(table_sources_index))
             table_list.append(table)
+        
+        #For Join Operator
+        if table_sources_count > 1: # Do Not Support More Than 2 Sources
+            left_child = table_list[0]
+            right_source = table_list[1]
+            
+            join_type = None 
+            if right_source.is_func_expr():
+                join_type = JoinType.LATERAL_JOIN
+            elif right_source.is_table_atom():
+                join_type = JoinType.HASH_JOIN
```

Comment:
Join can be one of these https://www.techonthenet.com/postgresql/joins.php. Don't call hash a separate type.